scipy/optimize/tests/test_minpack.py

```diff
@@ -551,6 +551,75 @@ def test_NaN_handling(self):
         assert_raises(ValueError, curve_fit, lambda x, a, b: a*x + b,
                       xdata, ydata, **{"check_finite": True})
 
+    @staticmethod
+    def _check_nan_policy(f, xdata_with_nan, xdata_without_nan,
+                          ydata_with_nan, ydata_without_nan, method):
+        # propagate test
+        error_msg = "`propagate` is not supported for nan_policy " \
+                    "in this function."
```

Comment:
If the suggestion above is accepted,
```suggestion
        error_msg = ("`nan_policy='propagate'` is not supported "
                    "by this function.")
```